schemachange/cli.py

```diff
@@ -271,12 +271,24 @@ def authenticate(self):
         warnings.warn(_warn_password, DeprecationWarning)
         snowflake_password = os.getenv("SNOWSQL_PWD")
 
-    if snowflake_password:
+    # if password and not okta authenticator, then default the authenticator
+    if snowflake_password and not os.getenv("SNOWFLAKE_AUTHENTICATOR").lower()[:8] == 'https://' \
```

Comment:
when the authenticator is not set, you cannot access the .lower() method because the object = None does not have a lower method.

The logic to determine authenticator should start with checking if authenticator is set or not, Then determine the type of authenticator.